oidc-service-impl/src/test/java/io/mosip/esignet/services/AuthorizationHelperServiceTest.java

```diff
@@ -284,6 +296,56 @@ public void validateAcceptedClaims_withInvalidAcceptedClaims_thenFail() {
         }
     }
 
+    @Test
+    public void validateAcceptedClaims_withValidAcceptedEssentialClaims_thenPass() {
+        Claims resolvedClaims = new Claims();
+        resolvedClaims.setUserinfo(new HashMap<>());
+        Map<String, ClaimDetail> userinfoClaims = new HashMap<>();
+        userinfoClaims.put("name", new ClaimDetail(null, null, true));
+        userinfoClaims.put("birthdate", new ClaimDetail(null, null, true));
+        userinfoClaims.put("address", new ClaimDetail(null, null, false));
+        userinfoClaims.put("gender", null);
+        resolvedClaims.setUserinfo(userinfoClaims);
+        OIDCTransaction oidcTransaction = new OIDCTransaction();
+        oidcTransaction.setRequestedClaims(resolvedClaims);
+        authorizationHelperService.validateAcceptedClaims(oidcTransaction, Arrays.asList("name", "birthdate"));
+    }
+
+    @Test
+    public void validateAcceptedClaims_withAllOptionalClaimsNotAccepted_thenPass() {
+        Claims resolvedClaims = new Claims();
+        resolvedClaims.setUserinfo(new HashMap<>());
+        Map<String, ClaimDetail> userinfoClaims = new HashMap<>();
+        userinfoClaims.put("name", new ClaimDetail(null, null, false));
+        userinfoClaims.put("birthdate", new ClaimDetail(null, null, false));
+        userinfoClaims.put("address", new ClaimDetail(null, null, false));
+        userinfoClaims.put("gender", null);
+        resolvedClaims.setUserinfo(userinfoClaims);
+        OIDCTransaction oidcTransaction = new OIDCTransaction();
+        oidcTransaction.setRequestedClaims(resolvedClaims);
+        authorizationHelperService.validateAcceptedClaims(oidcTransaction, List.of());
+    }
+
+    @Test
+    public void validateAcceptedClaims_withAllOptionalClaims_thenFail() {
+        Claims resolvedClaims = new Claims();
+        resolvedClaims.setUserinfo(new HashMap<>());
+        Map<String, ClaimDetail> userinfoClaims = new HashMap<>();
+        userinfoClaims.put("name", new ClaimDetail(null, null, false));
+        userinfoClaims.put("birthdate", new ClaimDetail(null, null, false));
+        userinfoClaims.put("address", new ClaimDetail(null, null, false));
+        userinfoClaims.put("gender", null);
+        resolvedClaims.setUserinfo(userinfoClaims);
+        OIDCTransaction oidcTransaction = new OIDCTransaction();
+        oidcTransaction.setRequestedClaims(resolvedClaims);
+        try {
+            authorizationHelperService.validateAcceptedClaims(oidcTransaction, Arrays.asList("email", "phone_number"));
```

Comment:
@vishwa-vyom  what should happen when unregistered/unrequested claims are sent as accepted claims in the auth-code request ? should we ignore? or throw an exception?

Reply:
Lets ignore and remove those claims from the final list

Reply:
@ase-101  @vishwa-vyom 
Sure I can modify the filter to perform this activity, but then the logic would look be something like this


- Filter the accepted claims to remove the elements which are not present in requested claims.
- All essential claims should be present in accepted claims , else  throw exception.

Have a few questions reg the workflow.

- if accepted claims contains something other than what's present in requested claims, wouldn't that imply in some issue in validation before. https://github.com/mosip/esignet/blob/d9fc5058ba03e128ddd6e0a1f87106ace68c3497/oidc-service-impl/src/main/java/io/mosip/esignet/services/AuthorizationServiceImpl.java#LL98C27-L98C36

- the consent screen in the end are shown by the esignet UI based on the oauth-details api called by the user

Reply:
This issue we are trying to solve is when the API is directly used @jainhitesh9998

Reply:
As per the latest discussion we are fine with thrown exceptions on this case to behave consistent with the other case where accepted claim does not contain all the essential claims